### PR TITLE
Ocamlformat 0.21.0 and Cmdliner 1.1.1

### PR DESCRIPTION
**Pull request by Julow**
###### Description of changes

New version of ocamlformat and cmdliner. Cmdliner update is tricky, see commit messages.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Julow** [comment]
Also, cmdliner's license changed in version 1.0.0. I've updated it

File: pkgs/development/ocaml-modules/cmdliner/1_1.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchurl, ocaml, findlib, ocamlbuild, topkg, result }:
+
+assert (lib.versionAtLeast ocaml.version "4.08");
```
Comment (symphorien) [inline pkgs/development/ocaml-modules/cmdliner/1_1.nix:3]: you can use minimalOCamlVersion instead for a better error message

**Review by symphorien — COMMENTED** [review]
diff looks good to me overall. could you fix the merge conflict ?

File: pkgs/development/ocaml-modules/cmdliner/1_1.nix (hunk 1 above)
Comment (Julow) [reply pkgs/development/ocaml-modules/cmdliner/1_1.nix]: Done, thanks.

**Comment by Julow** [comment]
I resolved the conflicts.

**Pull request merged**